### PR TITLE
cf-sketch: fix relocate_path and use it in the tools/test/Makefile

### PR DESCRIPTION
**Pull request by tzz**
This enables `relocate_path` again, making it possible to use DC with relocated inputs.

### CODE REVIEW

**Comment by tzz** [comment]
@nickanderson @estenberg I hope this is acceptable.  See discussion in mailing list.

**Comment by tzz** [comment]
I added a commit to remove `--test=any` from the test Makefile, since test mode was removed recently in #448

**Comment by tzz** [comment]
ping?

**Comment by nickanderson** [comment]
For reference here is the thread on the mailing list: https://groups.google.com/d/msg/help-cfengine/NN_NuVrsnvU/1b4qcBAnaeIJ

It seems fine to me, but I don't know the code well. @estenberg ?

**Comment by tzz** [comment]
ping

**Comment by tzz** [comment]
ping @estenberg

**Comment by estenberg** [comment]
Intention is good but I don't have much to say about the code..
Only thing I'm worrying about is breakage of existing DC usage.

File: tools/cf-sketch/cf-sketch.pl [hunk 1]
```diff
@@ -359,6 +354,11 @@ sub api_interaction
         $opts->{runfile}->{header} = $options{runfile_header};
     }
 
+    if (exists $options{relocate_path})
```
Comment (tzz) [inline tools/cf-sketch/cf-sketch.pl]: Note this is off by default.

File: tools/cf-sketch/perl-lib/DCAPI.pm [hunk 2]
```diff
@@ -1339,7 +1339,7 @@ sub regenerate
     my @inputs;
     foreach my $a (@activations)
     {
-        push @inputs, $a->sketch()->get_inputs(undef, 10);
+        push @inputs, $a->sketch()->get_inputs($self->runfile()->{relocate_path}, 10);
```
Comment (tzz) [inline tools/cf-sketch/perl-lib/DCAPI.pm]: This is `undef` by default.

**Comment by tzz** [comment]
The default behavior is same as before.  It's only changed if the user explicitly requests `relocate_path=/a/b/c/whatever` from the API.

**Comment by estenberg** [comment]
OK, what can go wrong... :)

**Pull request merged**